fact/build.rs

```diff
@@ -5,11 +5,14 @@ use std::{
     process::Command,
 };
 
-fn compile_bpf(out_dir: &Path) -> anyhow::Result<()> {
+fn compile_bpf(out_dir: &Path, arch: &str) -> anyhow::Result<()> {
```

Comment:
Do we need to pass `arch` as an argument? Seeing that we are reading it from an environment variable, I would assume we can just do this instead:
```rust
let target_arch = format!("-D__TARGET_ARCH_{}", env::var("CARGO_CFG_TARGET_ARCH")?);
```

If we were using this somewhere else, it might make more sense in my mind, but for the time being I think this would be more concise, WDYT?